sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java

```diff
@@ -129,7 +127,12 @@ public void recordGatewayResponse(
             }
             this.gatewayStatistics = new GatewayStatistics();
             if (rxDocumentServiceRequest != null) {
+                if (rxDocumentServiceRequest.getResourceType().equals(ResourceType.Document)) {
+                    connectionMode = ConnectionMode.GATEWAY;
```

Comment:
Doesn't the client configuration have this information? Why is it needed on the ClientSideRequestStatistics?

Reply:
Currently client configuration doesnt have connection mode ,  i was trying to make scope of this PR small, but let change in client configuration in this PR itself